odb-compiler/include/odb-compiler/ast/Block.hpp

```diff
@@ -14,6 +14,8 @@ class ODBCOMPILER_PUBLIC_API Block : public Node
 {
 public:
     Block(SourceLocation* location);
+    ~Block();
```

Comment:
Destructor doesn' do anything, I would just delete it

Reply:
Unfortunately, it's required for linking purposes.

`Block` contains `Reference<T>` where `T` is forward declared, so when `~Block` is generated by the compiler (forced by the dllexport), `Reference<T>::~Reference` will try and call a function on the forward declared type, then error.

The way around this is to declare the destructor, and define it to the default in the cpp file, where the `~Reference` destructor will compile correctly because `T` is no longer forward declared.